### PR TITLE
chore: move stat reporting into workspacestats package

### PR DESCRIPTION
**Pull request by f0ssel**
Part of https://github.com/coder/coder/issues/13279

- Removes `workspaceapps.StatsDBReporter` in favor of new `workspacestats.Reporter`
- Changes `workspaceapps.StatsReporter` interface method name for clarity
- `workspaceapps.StatsCollector` now uses `workspacestats.Reporter.ReportAppStats()`
- The agent API and legacy API routes now use `workspacestats.Reporter.ReportAgentStats()`

This is primarily just moving code, the only functional change here is that the legacy agent API routes previously only bumped `last_used_at` if there was a tracked "session connection"`, but this diverges from our new agent API where we always bump it. The behavior is now unified to always bump the field since the legacy API is on the way out.

### CODE REVIEW

File: coderd/workspaceapps/proxy.go [hunk 1]
```diff
@@ -573,7 +573,7 @@ func (s *Server) proxyWorkspaceApp(rw http.ResponseWriter, r *http.Request, appT
 	}
 
 	// This strips the session token from a workspace app request.
-	cookieHeaders := r.Header.Values("Cookie")[:]
+	cookieHeaders := r.Header.Values("Cookie")
```
Comment (johnstcn) [inline coderd/workspaceapps/proxy.go:576]: Is there a reason we want to avoid creating the new slice here?

**Review by johnstcn — APPROVED** [review]
LGTM, and the tests also appear to be happy.

It's nice that we now have unified behaviour between the legacy HTTP and dRPC agent APIs.

File: coderd/workspaceapps/proxy.go (hunk 1 above)
Comment (f0ssel) [reply coderd/workspaceapps/proxy.go:576]: I have no clue how this change even got in tbh, thanks for the catch

Comment (johnstcn) [reply coderd/workspaceapps/proxy.go]: Might be an automated linter thing

**Pull request merged**